core/src/network/reducers/get_links.rs

```diff
@@ -1,27 +1,23 @@
 use crate::{
-    action::ActionWrapper,
+    action::{ActionWrapper, GetLinksKey},
     context::Context,
     network::{reducers::send, state::NetworkState},
 };
-use holochain_core_types::{cas::content::Address, error::HolochainError};
+use holochain_core_types::error::HolochainError;
 use holochain_net_connection::protocol_wrapper::{GetDhtMetaData, ProtocolWrapper};
 use std::sync::Arc;
 
-fn inner(
-    network_state: &mut NetworkState,
-    address: &Address,
-    tag: &String,
-) -> Result<(), HolochainError> {
+fn inner(network_state: &mut NetworkState, key: &GetLinksKey) -> Result<(), HolochainError> {
     network_state.initialized()?;
 
     send(
         network_state,
         ProtocolWrapper::GetDhtMeta(GetDhtMetaData {
-            msg_id: "?".to_string(),
+            msg_id: key.id.clone(),
```

Comment:
This makes me uneasy. Now that I think about it, I'm wondering if we shouldn't add an extra field to GetDhtMetaData for this. With this fix we're hijacking msg_id to store data we actually care about, rather than just using it as an arbitrary UID. If we lose sight of this we'll break things again. What do you think? An extra "data" string field?

Reply:
Or am I over thinking this?

Reply:
Well, I thing this usage (a unique local (session?) ID) is what this field was intended for. Right @neonphog?

Reply:
If so then I approve!

Reply:
The networking module just clones this value into the response. It is intended for core to be able to link asynchronous request / responses.